website/projects/map4sci/src/app/map/map.component.ts

```diff
@@ -298,6 +295,10 @@ export class MapComponent {
 
   // Returns whether or not the zoom level has changed enough to warrant a change in which edges we are displaying.
   edgeLevelChange(): boolean {
+    if (!this.edges) {
```

Comment:
the getEdges (or setup as a proper getter) woudl help eliminate a lot of these guards around edges and ndoes.